File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.rb
# @param {String} s
# @param {Integer} k
# @return {Integer}
def max_vowels(s, k)

    vowels = "aeiou"
    string = s.chars
    sum = 0
    max = -Float::INFINITY

    string.each_with_index do |c, i|
        sum+= 1 if vowels.match?(c)
        if i + 1 >= k 
            if i >= k
                sum -= 1 if vowels.match?(string[i - k])
            end
            max = [sum,max].max
            return max if max == k
        end     
    end
    max
end

File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.ts
function maxVowels(s: string, k: number): number {
    let vowels = "aeiou"
    let count = 0
    let max = -Infinity

    for (let i = 0; i < s.length; i++){
        if (i >= k){
            //subtract previous
            if (vowels.includes(s[i - k])){
                count--
            }
        }
        if (vowels.includes(s[i])){
            count++
        }

        max = Math.max(count,max)
    }

    return max 
};